crates/lox-core/src/math/roots.rs

```diff
@@ -300,19 +303,21 @@ where
 
 #[cfg(test)]
 mod tests {
-    use std::f64::consts::PI;
-
     use lox_test_utils::assert_approx_eq;
+    use std::convert::Infallible;
+    use std::f64::consts::PI;
 
     use super::*;
 
+    pub type Fallible = Result<f64, Infallible>;
```

Comment:
```suggestion
    pub type Result = std::result::Result<f64, Infallible>;
```
This is a common idiom.